### PR TITLE
[2.0.2] zh: add `start-relay` and `stop-relay`

### PR DESCRIPTION
**Pull request by lance6716**
### Which DM version(s) do your changes apply to? (Required)

- [x] master (the latest development version, including v3.0 changes)
- [x] v2.0 (TiDB DM 2.0 versions)
- [ ] v1.0 (TiDB DM 1.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s): align with https://github.com/pingcap/dm/pull/1515

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: zh/relay-log.md [hunk 1]
```diff
@@ -6,9 +6,13 @@ aliases: ['/docs-cn/tidb-data-migration/dev/relay-log/']
 
 # DM Relay Log
 
-DM (Data Migration) 工具的 relay log 由一组有编号的文件和一个索引文件组成。这些有编号的文件包含了描述数据库更改的事件。索引文件包含所有使用过的 relay log 的文件名。
+DM (Data Migration) 工具的 relay log 由若干组有编号的文件和一个索引文件组成。这些有编号的文件包含了描述数据库更改的事件。索引文件包含所有使用过的 relay log 的文件名。
 
-DM-worker 在启动后，会自动将上游 binlog 迁移到本地配置目录（若使用 TiUP 部署 DM，则迁移目录默认为 `<deploy_dir> / relay_log` ）。DM-worker 在运行过程中，会将上游 binlog 实时迁移到本地文件。DM-worker 的 sync 处理单元会实时读取本地 relay log 的 binlog 事件，将这些事件转换为 SQL 语句，再将 SQL 语句迁移到下游数据库。
+在 v2.0.2 之前的版本（不含 v2.0.2），DM-worker 在绑定上游数据源时，会检查上游数据源配置中的 `enable-relay` 项。如果 `enable-relay` 为 `true`，则为该数据源启用 relay log 功能。
+
+在 v2.0.2 及之后的版本，`start-relay` 命令用于配置一个或多个 DM-worker 为指定数据源迁移 relay log。
```
Comment (CharLotteiu) [inline zh/relay-log.md:13]: 在 v2.0.2 及之后的版本中，为上游数据源配置 `enable-relay` 项还起作用吗？这个配置项是废弃掉了还是不推荐用了？建议增加一下这方面的说明。

File: zh/relay-log.md [hunk 2]
```diff
@@ -70,9 +74,17 @@ Relay log 本地存储的目录结构示例如下：
 
 ## 初始迁移规则
 
-DM-worker 每次启动时（或在 DM-worker 暂停后 relay log 恢复迁移），迁移的起始位置会出现以下几种情况：
+Relay log 迁移的起始位置由如下规则决定：
+
+- 从下游数据库 sync 单元 checkpoint 中，获取各同步任务需要该数据源的最早位置。如果该位置比下述任何一个位置更加新，则从此位置开始迁移。
```
Comment (CharLotteiu) [inline zh/relay-log.md:79]: 此处的`新`是指记录该位置的时间吗？

```suggestion
- 从下游数据库 sync 单元 checkpoint 中，获取各同步任务需要该数据源的最早位置。如果该位置的记录时间晚于下述任何一个位置，则从此位置开始迁移。
```

File: zh/relay-log.md [hunk 3]
```diff
@@ -70,9 +74,17 @@ Relay log 本地存储的目录结构示例如下：
...
+
+- 若本地 relay log 有效（有效是指 relay log 具有有效的 `server-uuid.index`，`subdir` 和 `relay.meta` 文件），DM-worker 从 `relay.meta` 记录的位置恢复迁移。
+
+- 若不存在有效的本地 relay log，但上游数据源配置文件中指定 `relay-binlog-name` 或 `relay-binlog-gtid`：
```
Comment (CharLotteiu) [inline zh/relay-log.md:83]:
```suggestion
- 若不存在有效的本地 relay log，但上游数据源配置文件中指定了 `relay-binlog-name` 或 `relay-binlog-gtid`：
```

File: zh/relay-log.md [hunk 4]
```diff
@@ -84,10 +96,96 @@ DM-worker 每次启动时（或在 DM-worker 暂停后 relay log 恢复迁移）
         >
         > 若上游的 relay log 被清理掉，则会发生错误。在这种情况下，需设置 `relay-binlog-gtid` 来指定迁移的起始位置。
 
-- 若不存在有效的本地 relay log：
+## 启动、停止 relay log
 
-    - 在非 GTID 模式下，若指定了 `relay-binlog-name`，则 DM-worker 从指定的 binlog 文件开始迁移。
-    - 在 GTID 模式下，若指定了 `relay-binlog-gtid`，则 DM-worker 从指定的 GTID 开始迁移。
+在 v2.0.2 及之后的版本，`start-relay` 与 `stop-relay` 命令可以分别启动及停止 relay log 的拉取。`start-relay` 命令只能指定空闲或者已绑定了该上游数据源的 DM-worker 拉取 relay log。
```
Comment (CharLotteiu) [inline zh/relay-log.md:101]:
```suggestion
在 v2.0.2 及之后的版本中，`start-relay` 与 `stop-relay` 命令可以分别启动及停止 relay log 的拉取。`start-relay` 命令只能指定空闲或者已绑定了该上游数据源的 DM-worker 拉取 relay log。
```

File: zh/relay-log.md (hunk 2 above)
Comment (lance6716) [reply zh/relay-log.md:79]: 也不是“记录时间”，而是一个逻辑上的新旧/早晚。可以改成“如果该位置晚于下述任何一个位置”？

Comment (CharLotteiu) [reply zh/relay-log.md:83]: OK. Suggestion updated.

**Review by yikeke — APPROVED** [review]

**Comment by GMHDBJD** [comment]
Warn 一下worker切换时如果没有 start-relay 也会关掉？

**Comment by lance6716** [comment]
> Warn 一下worker切换时如果没有 start-relay 也会关掉？

指的是 source 绑定到没开 relay 的 worker 上的话，就不会启用 relay 吗。我觉得挺自然的 😂

File: zh/relay-log.md [hunk 5]
```diff
@@ -6,9 +6,17 @@ aliases: ['/docs-cn/tidb-data-migration/dev/relay-log/']
 
 # DM Relay Log
 
-DM (Data Migration) 工具的 relay log 由一组有编号的文件和一个索引文件组成。这些有编号的文件包含了描述数据库更改的事件。索引文件包含所有使用过的 relay log 的文件名。
+DM (Data Migration) 工具的 relay log 由若干组有编号的文件和一个索引文件组成。这些有编号的文件包含了描述数据库更改的事件。索引文件包含所有使用过的 relay log 的文件名。
 
-DM-worker 在启动后，会自动将上游 binlog 迁移到本地配置目录（若使用 TiUP 部署 DM，则迁移目录默认为 `<deploy_dir> / relay_log` ）。DM-worker 在运行过程中，会将上游 binlog 实时迁移到本地文件。DM-worker 的 sync 处理单元会实时读取本地 relay log 的 binlog 事件，将这些事件转换为 SQL 语句，再将 SQL 语句迁移到下游数据库。
+在 v2.0.2 之前的版本（不含 v2.0.2），DM-worker 在绑定上游数据源时，会检查上游数据源配置中的 `enable-relay` 项。如果 `enable-relay` 为 `true`，则为该数据源启用 relay log 功能。
+
+在 v2.0.2 及之后的版本，`start-relay` 命令用于配置一个或多个 DM-worker 为指定数据源迁移 relay log。上游数据源配置中的 `enable-relay` 项已经失效。在[加载数据源配置](manage-source.md#加载数据源配置)时，如果发现配置中的 `enable-relay` 项为 `true`，会给出如下信息提示使用 `start-relay` 命令。
```
Comment (CharLotteiu) [inline zh/relay-log.md:13]: Should `[加载数据源配置](manage-source.md#加载数据源配置)` be changed according to the changes in https://github.com/pingcap/docs-dm/pull/605/?

**Comment by GMHDBJD** [comment]
> > Warn 一下worker切换时如果没有 start-relay 也会关掉？
> 
> 指的是 source 绑定到没开 relay 的 worker 上的话，就不会启用 relay 吗。我觉得挺自然的 😂 现在 relay 并不是 source 的一个属性，而是 worker 的一个属性。
> 
> 是这个意思的话我加一下

ok，现在这样也行

File: zh/relay-log.md (hunk 5 above)
Comment (lance6716) [reply zh/relay-log.md:13]: Yes, seems these two PRs has conflicts 😂 I'm going to change the bookmark in this PR, so we should let #605 get merged first

Comment (lance6716) [reply zh/relay-log.md]: https://github.com/pingcap/docs-dm/pull/608/commits/5f5445fe80b25d846dbc9deff6b3a2024efaa102

**Pull request merged**